.pre-commit-config.yaml

```diff
@@ -19,3 +19,16 @@ repos:
     rev: v13.0.0
     hooks:
       - id: clang-format
+#  - repo: local
+#    hooks:
+#      - id: clang-tidy
+#        name: clang-tidy
+#        description: Runs clang-tidy and checks for errors
+#        entry: python ./tools/pre-commit/clang-tidy.py
+#        language: python
+#        files: ^src/
+#        types: [c++, text]
+#        fail_fast: true
+#        require_serial: true
+#        args: [--compile_commands_path=build]
+#        pass_filenames: false
```

Comment:
The clang-tidy execution is SLOW. I want to add it to pre-commit for people to use, but need to openly talk to people before this is uncommented to impact every commit. 
The pre-commit stuff is thrown in here because IMO why not. If we are meant to keep PRs clean I deliver separately in future.